### PR TITLE
Theme JS linting

### PR DESCRIPTION
**Pull request by rob-howells**
**Note** - this is a follow up to https://github.com/accessdigital/wilson/pull/15, and so the target branch has been temporarily set to that PR. Once pull #15 is merged, the target will be updated.

**Background / Motivation**

* Currently there isn't any linting of the theme JS, presenting an opportunity to improve code standardisation 

**Approach**

* Added new gulp package ([gulp-eslint-new](https://www.npmjs.com/package/gulp-eslint-new)) and task to lint JS files
* Copied simplified version version of `estlintrc.json` from Drupal Core to use same set of lint rules, but removed any overly opinionated formatting plugins (i.e. Prettier) and unnecessary globals (e.g. jQuery) from the configuration
* Updated theme JS to adhere to new linting standards

**Additional work**

* Removed unused variables from gulpfile and converted functions to use ES6
* Moved button styles into Tailwind's `components` layer to avoid having to include button scss in the cookie scss

### CODE REVIEW

File: themes/wilson_theme_starterkit/sass/modules/_buttons.scss [hunk 1]
```diff
@@ -1,78 +1,80 @@
...
-  @apply transition-shadow;
-
-  &:hover,
-  &:focus,
-  a:hover &,
-  a:focus & {
-    @apply shadow-halo;
+@layer components {
+  .btn {
+    @apply inline-block;
```
Comment (iamsophiesk) [inline themes/wilson_theme_starterkit/sass/modules/_buttons.scss:3]: Oo. I've never seen `@apply` before.

File: themes/wilson_theme_starterkit/sass/modules/_buttons.scss [hunk 2]
```diff
@@ -1,78 +1,80 @@
...
-  @apply rounded-3xl;
-  @apply cursor-pointer;
-  @apply transition-shadow;
-
-  &:hover,
-  &:focus,
-  a:hover &,
-  a:focus & {
-    @apply shadow-halo;
+@layer components {
```
Comment (iamsophiesk) [inline themes/wilson_theme_starterkit/sass/modules/_buttons.scss:1]: Or `@layer`. I have some reading to do! :D

**Review by iamsophiesk — APPROVED** [review]
Is ESLint the preferred way of linting stuff? I might have to lift some of this for Wave, cos we don't really have anything solid in place right now, and it's annoying me.

But this looks good, can't beat a linting update :thumbsup:

File: themes/wilson_theme_starterkit/gulpfile.js [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+const path = require('path'),
+  sass = require('gulp-sass')(require('sass')),
+  sassGlob = require('gulp-sass-glob'),
+  cssnano = require('gulp-cssnano'),
+  rename = require('gulp-rename'),
+  postcss = require('gulp-postcss'),
+  terser = require('gulp-terser'),
+  gulpStylelint = require('gulp-stylelint'),
+  runSequence = require('gulp4-run-sequence'),
+  babel = require('gulp-babel');
```
Comment (AccessPaul) [inline themes/wilson_theme_starterkit/gulpfile.js:19]: Should this line end with a comma?

**Review by AccessPaul — APPROVED** [review]

File: themes/wilson_theme_starterkit/gulpfile.js (hunk 3 above)
Comment (rob-howells) [reply themes/wilson_theme_starterkit/gulpfile.js]: @AccessPaul - ah yes, updated

**Comment by rob-howells** [comment]
> Is ESLint the preferred way of linting stuff? I might have to lift some of this for Wave, cos we don't really have anything solid in place right now, and it's annoying me.
> 
> But this looks good, can't beat a linting update 👍

@iamsophiesk tends to be the go to in my experience but may be implemented slightly depending on the set up of Wave. Feel free to give me a shout when you get to implementing it if I can help with any pointers

**Pull request merged**